File: lib/Calculator/basic_calculator_2.py
def calculate(s) -> int:

    def helper(s) -> int:
        stack = []
        sign = '+'
        num = 0

        while len(s) > 0:
            c = s.pop(0)
            if c.isdigit():
                num = 10 * num + int(c)

            if (not c.isdigit() and c != ' ') or len(s) == 0:
                if sign == '+':
                    stack.append(num)
                elif sign == '-':
                    stack.append(-num)
                elif sign == '*':
                    stack[-1] = stack[-1] * num
                elif sign == '/':
                    # python 除法向 0 取整的写法
                    stack[-1] = int(stack[-1] / float(num))
                num = 0
                sign = c

        return sum(stack)
    # 需要把字符串转成列表方便操作
    return helper(list(s))

print(calculate("3+2"))

File: lib/Calculator/basic_calculator_2.js
function calculator(s) {

  function helper(s) {
    let stack = []
    let sign = "+"
    let num = 0
    let arr = Array.from(s)

    while(arr.length > 0) {
      let c = arr.shift()

      if(isDigit(c)) {
        num = 10 * num + (c - 0)
      }

      if((!isDigit(c) && c != " ") || arr.length == 0) {
        if(sign == "+") {
          stack.push(num)
        } else if(sign == "-") {
          stack.push(-num)
        } else if(sign == "*") {
          stack[stack.length - 1] *= num
        } else if(sign == "/") {
          let prev = stack[stack.length - 1]
          let result = prev / num

          result > 0
          ? stack[stack.length - 1] = Math.floor(result)
          : stack[stack.length - 1] = Math.ceil(result)
        }
        num = 0
        sign = c
      }
    }
    return sum(stack)
  }

  function sum(arr){
    return arr.reduce((accumulator, currentValue) => accumulator + currentValue)
  }

  function isDigit(str) {
    return /^\d+$/.test(str)
  }

  return helper(s)
}

// console.log(calculator("3*2"))
// console.log(calculator(" 3/2 "))
// console.log(calculator("3+5 / 2"))
